### ISSUE TITLE
Move eslint dep to peerDeps or remove it

### ISSUE THREAD

**Issue by bkniffler** [prettier/prettier-eslint#133]
Hey, 
Putting eslint to the dependencies can cause issues (like for me, I'm trying to use eslint 4.9.0 but prettier-eslint interferes)
```
"prettier-eslint": {
      "version": "8.2.0",
      "resolved": "https://registry.npmjs.org/prettier-eslint/-/prettier-eslint-8.2.0.tgz",
      "integrity": "sha512-OSx/0ULxWuIjeMO6KJoMVIDc7/Kj9a0wxUDzaGJTNiZXlNRq2bCl5jya8p42OQCVJzp1HnnQyGrWb+6/5+jbJQ==",
      "dev": true,
      "requires": {
        "common-tags": "1.4.0",
        "dlv": "1.1.0",
        "eslint": "4.8.0",
        "indent-string": "3.2.0",
        "lodash.merge": "4.6.0",
        "loglevel-colored-level-prefix": "1.0.0",
        "prettier": "1.7.3",
        "pretty-format": "20.0.3",
        "require-relative": "0.8.7",
        "typescript": "2.5.3",
        "typescript-eslint-parser": "7.0.0"
      },
      "dependencies": {
        "indent-string": {
          "version": "3.2.0",
          "resolved": "https://registry.npmjs.org/indent-string/-/indent-string-3.2.0.tgz",
          "integrity": "sha1-Sl/W0nzDMvN+VBmlBNu4NxBckok=",
          "dev": true
        },
        "lodash.merge": {
          "version": "4.6.0",
          "resolved": "https://registry.npmjs.org/lodash.merge/-/lodash.merge-4.6.0.tgz",
          "integrity": "sha1-aYhLoUSsM/5plzemCG3v+t0PicU=",
          "dev": true
        }
      }
    }
```

**Comment by judewang** [comment]
I have the same issue, too. I have no idea why prettier-eslint broke my lint test, now I got it😱

**Comment by zimme** [comment]
I believe we should have both eslint and prettier as peer dependencies, but I guess we need to lock it down to major versions at least.

**Comment by zimme** [comment]
I'm note sure if this is a dep vs peer dep problem or a dep versioning problem.

**Comment by zimme** [comment]
I've read up a bit on peerDependencies. I don't believe this is a peerDependency problem at all, it's just a versioning mismatch problem.

`prettier-eslint` isn't using `prettier` or `eslint` in any way that would require you to include them in your project explicitly. They are used under the hood by `prettier-eslint`, and because of this there shouldn't be a need to have them as peerDependencies.

I do believe that the problem comes from the fact that we don't actually use the versions of `prettier` and `eslint` that we specify in our dependencies. We're actually using the versions of the packages that are in the closest `node_modules` folder from the file that we're formatting.

I need to look into this a bit more and see what a good way forward would be.

I do believe @azz might have the same challenges in `prettier-tslint`, so hopefully we can try and figure this out together.

**Comment by zimme** [comment]
I don't see a reason to move either `prettier´ or `eslint` to `peerDependencies´.

Unless someone can give me a fully motivated reason for doing this I don't see this happening.

**Comment by mxmzb** [comment]
@zimme i will give it a shot: i just migrated a bunch of projects into a monorepo. one of these projects happens to be a `create-react-app` which comes with its own version of `eslint`. configuring anything at this point in CRA would require me to eject (or turn off the preflight check).

now when i wanted to move up all my dev dependencies from my projects that are used multiple times across the monorepo, i can't bring up `prettier-eslint`, because it will also install a newer version of eslint in my root `node_modules` and interfere with the CRA project.

tldr: moving `eslint` to peer dependencies could help working with mono repos (which i feel are increasingly receiving attention currently) but state no real disadvantages, besides of that devs need to have `eslint` declared explicitly in their `package.json` (which is also good practice imho and stimulates the understanding of devs who don't understand their `package.json` yet).

**Comment by aMarCruz** [comment]
yes, there's issues with CRA
